src/PIL/IcnsImagePlugin.py

```diff
@@ -391,8 +391,8 @@ def _accept(prefix):
     with open(sys.argv[1], "rb") as fp:
         imf = IcnsImageFile(fp)
         for size in imf.info["sizes"]:
-            imf.size = size
-            imf.save("out-%s-%s-%s.png" % size)  # noqa: UP031
+            w, h, x = imf.size = size
+            imf.save(f"out-{w}-{h}-{x}.png")
```

Comment:
```suggestion
            width, height, scale = imf.size = size
            imf.save(f"out-{width}-{height}-{scale}.png")
```
The third part is "scale", as seen under "sizes" at https://pillow.readthedocs.io/en/stable/handbook/image-file-formats.html#icns